pymeasure/adapters/prologix.py

```diff
@@ -144,42 +165,52 @@ def write_binary_values(self, command, values, **kwargs):
         """
         if self.address is not None:
             address_command = "++addr %d\n" % self.address
-            self.connection.write(address_command.encode())
-        super().write_binary_values(command, values, **kwargs)
-        self.connection.write(b'\n')
+            self.write(address_command)
+        super().write_binary_values(command, values, "\n", **kwargs)
 
-    def read(self):
-        """ Reads the response of the instrument until timeout
+    def _read(self, **kwargs):
+        """Read up to (excluding) `read_termination` or the whole read buffer.
 
-        :returns: String ASCII response of the instrument
+        :param kwargs: Keyword arguments for the connection itself.
+        :returns str: ASCII response of the instrument (excluding read_termination).
         """
         self.write("++read eoi")
-        return b"\n".join(self.connection.readlines()).decode()
+        return super()._read()
 
-    def gpib(self, address, rw_delay=None):
-        """ Returns and PrologixAdapter object that references the GPIB
+    def gpib(self, address, query_delay=0, **kwargs):
+        """ Returns an PrologixAdapter object that references the GPIB
         address specified, while sharing the Serial connection with other
         calls of this function
 
         :param address: Integer GPIB address of the desired instrument
-        :param rw_delay: Set a custom Read/Write delay for the instrument
+        :param query_delay: Set a custom Read/Write delay for the instrument
         :returns: PrologixAdapter for specific GPIB address
         """
-        rw_delay = rw_delay or self.rw_delay
-        return PrologixAdapter(self.connection, address, rw_delay=rw_delay)
+        query_delay = query_delay or kwargs.pop('rw_delay', 0) or self.query_delay
```

Comment:
Still "issues" to manage query delay at `Adapter` level :(

Reply:
As I said before, we have query_delay for the Adapter **and** the Instrument. I didn't change anything waiting for a comment of @bilderbuchi . I personally prefer keeping it at both levels as either device (Adapter or Instrument) might require a delay. That way, any instrument can use any adapter and the delay is always the lowest possible of the Adapter/Instrument combination and not the longest one.